Missing feature flags causes build errors

In the pinocchio template , there are some feature flags missing in the generated clients folder and that causes borsh build errors 

The `derive` feature flag for borsh is not enabled in the `Cargo.toml` at the project root

## Fix 

```
[workspace]
members = [
    "program",
    "clients/rust",
]
resolver = "2"

[workspace.package]
version = "0.0.1"
edition = "2021"

[workspace.lints.rust]
unused_imports = "deny"
dead_code = "warn"
unused_variables = "warn"
deprecated = "warn"
unexpected_cfgs = { level = "warn", check-cfg = [
    'cfg(target_os, values("solana"))',
] }

[workspace.dependencies]
codama = "^0.7.2"
serde_json = "^1.0.145"
const-crypto = "^0.3.0"
pinocchio = { version = "^0.10.1", features = ["copy"] }
solana-address = { version = "2.0", features = ["curve25519"] }
pinocchio-log = "^0.5.1"
pinocchio-system = "^0.5.0"
pinocchio-token = "0.5.0"
thiserror = "^2.0.17"
borsh = { version = "^1.6.0", features = ["derive"] } // Enable feature flag!
num-derive = "^0.4.0"
num-traits = "^0.2.0"

```

in `clients/rust/Cargo.toml` the `borsh` feature for `solana_pubkey` is not enabled so it causes build errors when trying to run tests that uses the generated rust clients 

## Error 
```rust
error[E0277]: the trait bound `Pubkey: BorshSerialize` is not satisfied
  --> clients/rust/src/generated/accounts/config.rs:12:10
   |
12 | #[derive(BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
   |          ^^^^^^^^^^^^^^ the trait `BorshSerialize` is not implemented for `Pubkey`
   |
   = help: the following other types implement trait `BorshSerialize`:
             &T
             ()
             (T0, T1)
             (T0, T1, T2)
             (T0, T1, T2, T3)
             (T0, T1, T2, T3, T4)
             (T0, T1, T2, T3, T4, T5)
             (T0, T1, T2, T3, T4, T5, T6)
           and 87 others
```

## Fix
```rust
[package]
name = "myproject-client"
version = "0.0.1"
edition = "2021"
description = "Rust client for the Myproject Program"
license = "MIT"

[dependencies]
borsh = { workspace = true }
solana-account-info = "3.1.0"
solana-pubkey = { version = "4.0.0", features = ["borsh"] } //enable feature flag 
solana-address = "2.0.0"
solana-instruction = "3.1.0"
solana-cpi = "3.0.1"
num-derive = { workspace = true }
num-traits = { workspace = true }

solana-client = { version = "3.1.5", optional = true }
solana-account = { version = "~3.2", optional = true }
solana-program-error = "3.0.0"
thiserror = { workspace = true }

[features]
default = []
fetch = ["solana-client", "solana-account"]

[package.metadata.cargo-machete]
ignored = ["num-traits"]

```

--- a/pinocchio/pinocchio-counter/Cargo.toml
+++ b/pinocchio/pinocchio-counter/Cargo.toml
@@ -28,6 +28,6 @@ solana-address = { version = "2.0", features = ["curve25519"] }
 pinocchio-log = "^0.5.1"
 pinocchio-system = "^0.5.0"
 thiserror = "^2.0.17"
-borsh = "^1.6.0"
+borsh = { version = "^1.6.0", features = ["derive"] }
 num-derive = "^0.4.0"
 num-traits = "^0.2.0"

--- a/pinocchio/pinocchio-counter/clients/rust/Cargo.toml
+++ b/pinocchio/pinocchio-counter/clients/rust/Cargo.toml
@@ -8,7 +8,7 @@ license = "MIT"
 [dependencies]
 borsh = { workspace = true }
 solana-account-info = "3.1.0"
-solana-pubkey = "4.0.0"
+solana-pubkey = { version = "4.0.0", features = ["borsh"] }
 solana-address = "2.0.0"
 solana-instruction = "3.1.0"
 solana-cpi = "3.0.1"

